### PR TITLE
feat(helm): Support for Helm Chart.yaml appVersion field defaulting to project version

### LINKED ISSUE #2472: Add appVersion to the generated Chart.yaml file
### Component

Kubernetes Maven Plugin

### Is your enhancement related to a problem? Please describe

I would like to get the appVersion added to the chart, so that when I do

```bash
 helm list -n dev
```

When I do a simple:

```
./mvnw -pl microservice-app k8s:resource k8s:helm
```

The generated `Chart.yaml` contains a `version` key, but no `appVersion` key

```yaml
apiVersion: v1
name: microservice-app
<snip>
version: "0.3.1"
<snip>
```

Output is 
```
NAME            	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART                 	APP VERSION
microservice-app	dev      	1       	2023-11-23 10:58:51.765527 +0100 CET	deployed	microservice-app-0.3.1	0.3.1
```

Rather than

```
NAME            	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART                          	APP VERSION
microservice-app	dev      	1       	2023-11-23 10:32:28.876677 +0100 CET	deployed	microservice-app-0.3.1

```

This was mentioned in the following issue 3 years ago, at the time when helm 2 apparently didn't have `appVersion` https://github.com/eclipse/jkube/issues/113 

### Describe the solution you'd like

I would like the `appVersion` to be automatically added with the project version (same behaviour as `version`) 

### Describe alternatives you've considered

Another option is to have Chart resource fragments that allow adding a `Chart.helm.yml` in `src/main/jkube`.
And if this file contains
```yaml
appVersion: ${project.version}
```
then this gets templated and added to the output `Chart.yml`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by manusa**
## Description

Fix #2472

Adds the appVersion field to the HelmConfig class so that it can be propagated from the Helm plugin configuration and the `jkube.helm.appVersion` property.

The field is now provided by default with the current project version (i.e. by default Chart.yaml appVersion and version default to the same value)

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2473 (_2023-11-24T11:55:21Z_)

:gear: [JKube E2E Tests (6980415248)](https://github.com/jkubeio/ci/actions/runs/6980415248) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2473)](https://github.com/jkubeio/ci/actions/runs/6980415248/job/18995614171)

**Review by sunix — APPROVED** [review]

File: jkube-kit/doc/src/main/asciidoc/inc/helm/_jkube_helm.adoc [hunk 1]
```diff
@@ -101,10 +101,18 @@ endif::[]
...
 
 | *icon*
-| The Chart URL to an SVG or PNG image to be used as an icon , default is extracted from the kubernetes manifest
+| The Chart URL to an SVG or PNG image to be used as an icon, default is extracted from the kubernetes manifest
 (`kubernetes.yml`) `jkube.eclipse.org/iconUrl` annotation if not provided.
 | `jkube.helm.icon`
 
+| *appVersion*
+| The version of the application that Chart contains
+(`kubernetes.yml`) `jkube.eclipse.org/iconUrl` annotation if not provided.
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/helm/_jkube_helm.adoc:110]:
```suggestion
version of the application
```

**Comment by rohanKanojia** [comment]
E2E tests need to be updated

File: jkube-kit/doc/src/main/asciidoc/inc/helm/_jkube_helm.adoc (hunk 1 above)
Comment (manusa) [reply jkube-kit/doc/src/main/asciidoc/inc/helm/_jkube_helm.adoc:110]: :flushed: not sure how this ended up this way

**Review by rohanKanojia — APPROVED** [review]

**Comment by manusa** [comment]
> E2E tests need to be updated

I'll update the e2e tests once I merge this one, I think it'll be easier.

**Pull request merged**